### PR TITLE
GraphQL API

### PR DESCRIPTION
**Pull request by illegalprime**
Can merge!

### CODE REVIEW

**Comment by illegalprime** [comment]
Merge #155 first.

**Comment by illegalprime** [comment]
note: need better searching feature

**Comment by illegalprime** [comment]
note: also maybe rename `user` query to `find_user` so there's no `user { user ` in checkin

File: server/branch.ts [hunk 1]
```diff
@@ -213,7 +231,7 @@ export class BranchConfig {
 			await this.loadAllBranches();
 			return true;
 		}
-		catch {
+		catch (err) {
```
Comment (petschekr) [inline server/branch.ts]: TypeScript 2.5.x supports this type of try / catch syntax if you don't need the error variable

File: server/tsconfig.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
         ],
         "target": "es6",
         "lib": [
+            "esnext.asynciterable",
```
Comment (petschekr) [inline server/tsconfig.json]: I've never seen this library flag used before. What does it enable?

Comment (illegalprime) [reply server/tsconfig.json]: It enables a new-ish feature `asynciterable` which is an iterator that has an `async` `next()` method, so you can iterate over things that return promises.

**Comment by illegalprime** [comment]
@petschekr `apollo-codegen` only supports client-side code, see apollographql/graphql-tools#404.

**Comment by petschekr** [comment]
Ah ok I didn't realize that
I'll merge this now

**Pull request merged**